client/src/components/Query/Filter/index.jsx

```diff
@@ -26,49 +27,28 @@ export const createFilter = type => ({
 class Filter extends React.Component {
   constructor(props) {
     super(props);
+    const { dataSet = [], autoOpen, visible, sortData } = props;
     this.state = {
-      type: null,
-      data: null,
-      dataSet: null,
-      draft: null,
-      visible: null,
-      selected: false,
-      opened: null,
-      allOptions: null,
-      selection: [],
-      size: null,
-      page: null,
+      dataSet,
+      visible,
+      opened: autoOpen,
+      allOptions: cloneDeep(sortData),
+      size: 10,
+      page: 1,
     };
 
     this.isEditable = this.isEditable.bind(this);
     this.isRemovable = this.isRemovable.bind(this);
     this.isSelectable = this.isSelectable.bind(this);
-    this.isSelected = this.isSelected.bind(this);
     this.isOpened = this.isOpened.bind(this);
     this.isVisible = this.isVisible.bind(this);
     this.serialize = this.serialize.bind(this);
     this.handleClose = this.handleClose.bind(this);
-    this.handleSelect = this.handleSelect.bind(this);
     this.handleApply = this.handleApply.bind(this);
     this.handleCancel = this.handleCancel.bind(this);
     this.toggleMenu = this.toggleMenu.bind(this);
     this.handlePageChange = this.handlePageChange.bind(this);
     this.handleSearchByQuery = this.handleSearchByQuery.bind(this)
-
-    // @NOTE Initialize Component State
```

Comment:
What/where is the "default" state of this component?

Reply:
Like I explained this morning. Line 31 to 38.

Reply:
Ok, I understand. Logic moved from the constructor and taken from props.